install_files/securedrop-app-code/DEBIAN/preinst

```diff
@@ -13,12 +13,46 @@ set -e
 # for details, see http://www.debian.org/doc/debian-policy/ or
 # the debian-policy package
 
+
+function permanently_disable_swap() {
+    # Swap usage is prohibited in the context of SecureDrop, due to risk of
+    # forensic discovery recovering plaintext submissions that were written
+    # to disk prior to encrypting. If a swapfile is active, disable it,
+    # and disable all swap entries from /etc/fstab, to prevent mounting
+    # after reboots.
+
+    # Check if swapfile is currently enabled.
+    swap_result=$(swapon --summary)
+
+    # Search for path format in output.
+    swap_partition=$(grep '/' <<< "${swap_result}" | perl -ane 'print $F[0]')
```

Comment:
See previous comment. Also, this doesn't account for the case there's more than one swap partition (unlikely, but I don't want to discredit it, and I don't think supporting that scenario should significantly complicate this script). In fact, if there are multiple swap partitions, this command will return a string of their concatenated mapped paths. Then the `shred` command below will error out.